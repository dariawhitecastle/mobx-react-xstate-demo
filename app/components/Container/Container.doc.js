import { describe, PropTypes } from 'react-desc'

export const doc = Container => {
  const DocumentedContainer = describe(Container)
    .description('The top level Container container.')
    .usage(
      `import { Container } from 'component/Container';
<Container>...</Container>`,
    )
    .intrinsicElement('div')

  DocumentedContainer.propTypes = {
    full: PropTypes.bool.description('Whether to take the whole viewport.').defaultValue(false),
    plain: PropTypes.bool
      .description(
        'Whether or not Container should apply a global font-family, font-size, and line-height.',
      )
      .defaultValue(false),
    cssVars: PropTypes.bool
      .description('Whether to expose the css variables.')
      .defaultValue(false),
    theme: PropTypes.object.description('Custom styles for Container app component.'),
    userAgent: PropTypes.string.description(
      'User agent used to detect the device width for setting the initial breakpoint.',
    ),
  }

  return DocumentedContainer
}

export const themeDoc = {
  'container.extend': {
    description: 'Any additional style for Container.',
    type: 'string | (props) => {}',
    defaultValue: undefined,
  },
  'global.font.face': {
    description: 'Custom font face declaration',
    type: 'string | (props) => {}',
    defaultValue: undefined,
  },
}
