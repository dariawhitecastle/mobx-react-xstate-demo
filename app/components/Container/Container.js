/* eslint-disable react/prop-types */
/**
 *
 * Container
 *
 */

import React from 'react'
import { Grommet } from 'grommet'
import { doc } from './Container.doc'
import helpers from '../../utils/helpers'

function createContainer({ cssVars, children, full, plain, theme, userAgent }) {
  return (
    <Grommet cssVars={cssVars} full={full} plain={plain} theme={theme} userAgent={userAgent}>
      {children}
    </Grommet>
  )
}

export default helpers.createWithDoc({
  docFunction: doc,
  component: createContainer,
})
