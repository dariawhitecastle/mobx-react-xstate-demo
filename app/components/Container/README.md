<!-- ! This is a generated file. To make changes, edit <Component>.doc.js ! -->
## createContainer
The top level Container container.

## Usage

```javascript
import { Container } from 'component/Container';
<Container>...</Container>
```

## Properties

**full**

Whether to take the whole viewport.

```
boolean
```

**plain**

Whether or not Container should apply a global font-family, font-size, and line-height.

```
boolean
```

**cssVars**

Whether to expose the css variables.

```
boolean
```

**theme**

Custom styles for Container app component.

```
object
```

**userAgent**

User agent used to detect the device width for setting the initial breakpoint.

```
string
```
  
## Intrinsic element

```
div
```
## Theme
  
**container.extend**

Any additional style for Container. Expects `string | (props) => {}`.

Defaults to

```
undefined
```

**global.font.face**

Custom font face declaration Expects `string | (props) => {}`.

Defaults to

```
undefined
```
