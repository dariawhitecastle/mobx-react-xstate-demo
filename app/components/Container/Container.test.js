/**
 *
 * Tests for Container
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import { render } from '@testing-library/react'
import 'jest-styled-components'

import Container from './Container'

const children = <h1>Test</h1>
const renderComponent = (props = {}) => render(<Container {...props}>{children}</Container>)

test('it renders and matches snapshot', () => {
  const { container } = renderComponent()
  expect(container).toMatchSnapshot()
})
