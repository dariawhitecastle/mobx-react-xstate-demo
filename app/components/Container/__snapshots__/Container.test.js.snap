// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders and matches snapshot 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div>
  <div
    class="c0"
  >
    <h1>
      Test
    </h1>
  </div>
</div>
`;
